crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -735,6 +748,24 @@ impl<'w, 's> Commands<'w, 's> {
     pub fn add<C: Command>(&mut self, command: C) {
         self.push(command);
     }
+
+    /// Sends a "global" [`Trigger`] without any targets.
+    pub fn trigger(&mut self, event: impl Event) {
+        self.add(TriggerEvent { event, targets: () });
+    }
+
+    /// Sends a [`Trigger`] with the given `targets`.
+    pub fn trigger_targets(&mut self, event: impl Event, targets: impl TriggerTargets) {
+        self.add(TriggerEvent { event, targets });
+    }
+
+    /// Spawn an [`Observer`] and returns it's [`Entity`].
+    pub fn observe<E: Event, B: Bundle, M>(
```

Comment:
This and all the functions on `Entity(Mut)Ref`s should probably NOT store the observer on the entity and instead spawn a new entity. Otherwise the only way to have multiple observers for the same event type will be to spawn observers manually instead of this short hand API.

Plugin authors can easily overwrite observers from other authors & users could overwrite observers from plugin authors.

Reply:
Fully agreed. This was intended to spawn a new observer entity. Just a mistake on my part during the porting process.